File: javascript/test/Structural/Decorator.js
(function() {
  define(['Structural/Decorator/TeaLeaves', 'Structural/Decorator/ChaiDecorator'], function(TeaLeaves, ChaiDecorator) {
    'use strict';
    return describe('Test Structural Decorator', function() {
      return it('Can we brew some chai?', function() {
        var chaiDecorator, teaLeaves;
        teaLeaves = new TeaLeaves();
        chaiDecorator = new ChaiDecorator(teaLeaves);
        return expect(chaiDecorator.steepTea()).toEqual("bay leaf is steeping ,cinnamon stick is steeping ,ginger is steeping ,honey is steeping ,soy milk is steeping ,vanilla bean is steeping , tea is steeping with chai");
      });
    });
  });

}).call(this);

File: java/t/test/Structural/Decorator.java
//TestChaiDecorator.java - testing the decorator


package test.Structural;

import static org.junit.Assert.assertEquals;

import org.junit.Test;
import org.junit.Ignore;
import org.junit.runner.RunWith;
import org.junit.runners.JUnit4;

import Structural.Decorator.*;




public class Decorator {            
   
	@Test 
   public void thetester() {
       Tea teaLeaves = new TeaLeaves();
       Tea chaiDecorator = new ChaiDecorator(teaLeaves);
       org.junit.Assert.assertEquals("Decorator test",
				"bay leaf is steeping, cinnamon stick is steeping, ginger is steeping, honey is steeping, soy milk is steeping, vanilla bean is steeping",
				chaiDecorator.steepTea() );
   }
}
